pandas/tests/frame/test_duplicates.py

```diff
@@ -263,6 +263,19 @@ def test_drop_duplicates_tuple():
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize('df', [
+    DataFrame(),
+    DataFrame(columns=[]),
+    DataFrame(columns=['A', 'B', 'C']),
+    DataFrame(index=[]),
+    DataFrame(index=['A', 'B', 'C'])
+])
+def test_drop_duplicates_empty(df):
+    # GH 20516
+    result = df.drop_duplicates()
+    tm.assert_frame_equal(result, df)
```

Comment:
can you add a test case with `inplace=True` please?

Reply:
Sure.